nixos/modules/services/networking/networkmanager.nix

```diff
@@ -379,6 +380,52 @@ in {
           for more details.
         '';
       };
+      ensure-profiles = {
+        profiles = mkOption {
+          type = types.attrsOf ini.type;
+          default = { };
+          example = {
+            home-wifi = {
+              connection = {
+                id = "home-wifi";
+                type = "wifi";
+                permissions = "";
+              };
+              wifi = {
+                mac-address-blacklist = "";
+                mode = "infrastructure";
+                ssid = "Home Wi-Fi";
+              };
+              wifi-security = {
+                auth-alg = "open";
+                key-mgmt = "wpa-psk";
+                psk = "$PASSWORD";
```

Comment:
There is a little name-clash here because the environment file will contain

$WORK_PASSWORD
$HOME_PASSWORD
$HOTEL1_PASSWORD

etct, right? Maybe then $PASSWORD is not an optimal example?

Reply:
An alternative would be one environment file per profile. Not sure …

Reply:
> etct, right? Maybe then $PASSWORD is not an optimal example?

yeah you are right, I'll change that. 

> An alternative would be one environment file per profile. Not sure …

then we would need one unit per connection which doesn't seem ideal